src/pin.ts

```diff
@@ -0,0 +1,339 @@
+      'frontend/model/contracts',
+      'contracts',
+      'src/contracts'
+    ]
+
+    let foundContracts = false
+
+    for (const searchPath of commonPaths) {
+      const fullPath = join(this.projectRoot, searchPath)
+      if (existsSync(fullPath)) {
+        try {
+          const files = await readdir(fullPath)
+          const manifestFiles = files.filter(f => f.endsWith('.manifest.json'))
+
+          if (manifestFiles.length > 0) {
+            console.log(colors.blue(`📁 ${searchPath}/`))
+            for (const file of manifestFiles) {
+              const contractName = file.replace('.manifest.json', '')
+              const relativePath = join(searchPath, file)
+              console.log(`  ${colors.green(contractName)} - ${colors.gray(relativePath)}`)
+            }
+            console.log()
+            foundContracts = true
+          }
+        } catch {
+          // Ignore errors reading directories
+        }
+      }
+    }
+
+    if (!foundContracts) {
+      console.log(colors.yellow('No contract files found in common locations.'))
+      console.log(colors.gray('Make sure your contract files end with .js and are in a contracts/ directory.'))
+    }
+
+    // Show current pinned contracts if any exist
+    if (Object.keys(this.cheloniaConfig.contracts).length > 0) {
+      console.log(colors.cyan('📌 Currently pinned contracts:'))
+      for (const [name, config] of Object.entries(this.cheloniaConfig.contracts)) {
+        console.log(`  ${colors.green(name)} - ${colors.blue(config.version)} (${colors.gray(config.path)})`)
+      }
+    }
+  }
+
+  /**
+   * Create the directory structure for a contract version
+   * Structure: contracts/<contractName>/<version>/
+   */
+  private async createVersionDirectory (contractName: string, version: string) {
+    const versionDir = join(this.projectRoot, 'contracts', contractName, version)
+
+    console.log(colors.blue(`📁 Creating directory: contracts/${contractName}/${version}/`))
+    await mkdir(versionDir, { recursive: true })
+  }
+
+  /**
+   * Copy contract files to the version directory
+   * This includes both the main contract and its slim version if it exists
+   */
+  private async copyContractFiles (contractPath: string, contractName: string, version: string) {
+    const sourceDir = dirname(join(this.projectRoot, contractPath))
+    const targetDir = join(this.projectRoot, 'contracts', contractName, version)
+
+    // Copy main contract file
+    const mainFile = `${contractName}.js`
+    const mainSource = join(sourceDir, mainFile)
+    const mainTarget = join(targetDir, mainFile)
+
+    if (existsSync(mainSource)) {
+      await copyFile(mainSource, mainTarget)
+      console.log(colors.green(`✅ Copied ${mainFile}`))
+    } else {
+      throw new Error(`Main contract file not found: ${mainSource}`)
+    }
+
+    // Copy slim version if it exists
+    const slimFile = `${contractName}-slim.js`
+    const slimSource = join(sourceDir, slimFile)
```

Comment:
This information is in the manifest file, so there's no need to make assumptions about what the file name is.